pandas/core/dtypes/dtypes.py

```diff
@@ -2271,6 +2271,9 @@ def construct_from_string(cls, string: str) -> ArrowDtype:
                     "instance with specific parameters."
                 ) from err
             raise TypeError(f"'{base_type}' is not a valid pyarrow data type.") from err
+        except NameError as err:
```

Comment:
I think this is a fairly fragile fix if a refactor happens. Somewhere `import_optional_dependency` should be used instead